tests/e2e/specs/miscellaneous/WorkpaseUsingParent.spec.ts

```diff
@@ -0,0 +1,99 @@
+	loginTests.loginIntoChe();
+	let podName: string = '';
+
+	test('Create a workspace using a parent', async function (): Promise<void> {
+		const factoryUrl: string = `${BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL}/dashboard/#https://github.com/testsfactory/parentDevfile`;
+		await dashboard.waitPage();
+		await browserTabsUtil.navigateTo(factoryUrl);
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+		registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+		// sometimes the trust dialog does not appear at first time, for avoiding this problem we send click event for activating
+		await new Workbench().click();
+		await projectAndFileTests.performTrustAuthorDialog();
+	});
+
+	test('Check cloning of the test project', async function (): Promise<void> {
+		const expectedProjectItems: string[] = ['.devfile.yaml', 'parent.yaml', 'README.md'];
+		const visibleContent: ViewSection = await new SideBarView().getContent().getSection('parentdevfile');
+
+		for (const expectedProjectItem of expectedProjectItems) {
+			const visibleItem: ViewItem | undefined = await visibleContent.findItem(expectedProjectItem);
+			expect(visibleItem).not.equal(undefined);
+		}
+	});
+
+	test('Check devfile VS Code tasks ', async function (): Promise<void> {
+		const input: QuickOpenBox | InputBox = await new Workbench().openCommandPrompt();
+		await input.setText('>Tasks: Run Task');
+		const runTaskItem: QuickPickItem | undefined = await input.findQuickPick('Tasks: Run Task');
+		await runTaskItem?.click();
+		const devFileTask: QuickPickItem | undefined = await input.findQuickPick('devfile');
+		await devFileTask?.click();
+		const firstExpectedQuickPick: QuickPickItem | undefined = await input.findQuickPick('1. This command from the devfile');
+		const secondExpectedQuickPick: QuickPickItem | undefined = await input.findQuickPick('2. This command from the parent');
+		expect(firstExpectedQuickPick).not.eqls(undefined);
+		expect(secondExpectedQuickPick).not.eqls(undefined);
+	});
+
+	test('Check expected containers in the parent POD', function (): void {
+		const getPodNameCommand: string = `${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} get pods --selector=controller.devfile.io/devworkspace_name=sample-using-parent --output jsonpath=\'{.items[0].metadata.name}\'`;
+
+		podName = executeArbitraryShellScript(getPodNameCommand);
+		const containerNames: string = executeArbitraryShellScript(
+			`${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} get pod ${podName} --output jsonpath=\'{.spec.containers[*].name}\'`
+		);
+		expect(containerNames).contain('tools');
+		expect(containerNames).contains('che-gateway');
+
+		const initContainerName: string = executeArbitraryShellScript(
+			`${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} get pod ${podName} --output jsonpath=\'{.spec.initContainers[].name}\'`
```

Comment:
Can we use is it getPodAndContainerNames() here also? mb with some updates?

Reply:
So it makes sense to extract it if we use it regularly. But WorkpaseUsingParent.spec.ts test is pretty specific. These specific commands are used in this test. But if we get precedent to use the same command in future tests, it will be better to extract in a method.